src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeMultiFetch.scala

```diff
@@ -0,0 +1,164 @@
+package tech.cryptonomic.conseil.tezos
+
+import cats.{Id, Apply, Monad, Traverse}
+import cats.data.Kleisli
+import cats.syntax.semigroupal._
+import com.typesafe.scalalogging.LazyLogging
+import tech.cryptonomic.conseil.util.JsonUtil
+import tech.cryptonomic.conseil.util.JsonUtil.{JsonString, adaptManagerPubkeyField}
+import TezosTypes._
+
+/** Contract for a generic combination of operations that get json from the node,
+  * then converts said json to a value, all doing multiple requests together.
+  * The returned values are tupled with the corresponding input
+  * The interface assumes some kind of higher-order typed effect `F[_]` is part
+  * of the operation execution, and represents everything parametrically, to leave
+  * room for future evolutions of the code protocol.
+  * Thus the required implementation is in the form of a `cats.data.Kleisli` type,
+  * parametric on effect `F`, input `In`, output `Out`. As needed, the in/out types are
+  * wrapped in a `T[_]` type constructor that represents multiple values (e.g. a collection type).
+  *
+  * @tparam F the higher-order effect type (e.g. `Try`, `Future`, `IO`, `Either`)
+  * @tparam T a traversable type (used to wrap mutliple requests and corresponding return values)
+  */
+trait TezosNodeMultiFetch[F[_], T[_]] {
+  /** the input type*/
+  type In
+  /** the output type*/
+  type Out
+  /** the json representation type used*/
+  type JsonType
+
+  /** an effectful function from a collection of inputs `T[In]`
+    * to the collection of json, tupled with the corresponding input `T[(In, JsonType)]`
+    */
+  def fetchJsonBatch: Kleisli[F, T[In], T[(In, JsonType)]]
+
+  /** an effectful function that decodes the json value to an output `Out` */
+  def decodeJson: Kleisli[F, JsonType, Out]
```

Comment:
This is ambiguous to me. If In is the raw input type from the Tezos Node, and Json type is the representation of the parsing, what would an example of Out be?

Reply:
`In` is our actual input needed to call a single tezos endpoint (e.g. the offset from a specific block, when you call the `blocks` endpoint)

the `JsonType` now renamed, is the representation used for json, in this case `JsonType = String`, very simply